js/src/favorite-domain.ts

```diff
@@ -88,16 +89,21 @@ export class FavouriteDomain {
  */
 export const getFavoriteDomain = async (
   connection: Connection,
-  owner: PublicKey
+  owner: PublicKey,
 ) => {
   const [favKey] = FavouriteDomain.getKeySync(
     NAME_OFFERS_ID,
-    new PublicKey(owner)
+    new PublicKey(owner),
   );
 
   const favorite = await FavouriteDomain.retrieve(connection, favKey);
 
   const reverse = await reverseLookup(connection, favorite.nameAccount);
+  const domainOwner = await resolve(connection, reverse);
 
-  return { domain: favorite.nameAccount, reverse };
+  return {
+    domain: favorite.nameAccount,
+    reverse,
+    stale: !owner.equals(domainOwner),
```

Comment:
@dr497 does this represent the case when some domain is the user's favorite, but the domain itself is actually sold to someone else?

Reply:
Yes, it's to handle the case where the user does not own the domain anymore